internal/aws/containerinsight/httpclient/httpclient.go

```diff
@@ -0,0 +1,131 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpClient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const (
+	defaultMaxRetries              = 3
+	defaultTimeout                 = 1 * time.Second
+	defaultBackoffRetryBaseInMills = 200
+	maxHttpResponseLength          = 5 * 1024 * 1024 // 5MB
+)
+
+type HttpClient struct {
+	maxRetries              int
+	backoffRetryBaseInMills int
+	client                  clientProvider
+}
+
+type HttpClientProvider interface {
+	Request(path string, ctx context.Context, logger *zap.Logger) ([]byte, error)
```

Comment:
```suggestion
	Request(ctx context.Context, path string, logger *zap.Logger) ([]byte, error)
```

`ctx` should always be the first argument.

Reply:
Will change it, thanks